test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -2531,79 +2555,123 @@ class test_ucp_sockaddr_protocols_err_sender
+            ucs_status_t status;
+            while ((status = ucp_request_check_status(close_req)) ==
+                    UCS_INPROGRESS) {
+                ucp_worker_progress(sender_worker);
+            }
+            ASSERT_EQ(UCS_ERR_CANCELED, status);
+        }
 
-    /* Allow receiver to get RTS notification, but do not receive message
-     * body */
-    message = message_wait(receiver(), 0, 0, &info);
-    ASSERT_NE((void*)NULL, message);
-    ASSERT_EQ(UCS_INPROGRESS, ucp_request_check_status(sreq));
+        if (reg_mem) {
+            /* Check that the send request would not be completed until the
+             * memory handle for the same send buffer will not be released */
+            ucs_status_t status;
+            for (int i = 0;
+                 ((status = ucp_request_check_status(sreq)) ==
+                        UCS_INPROGRESS) && (i < 100); i++) {
+                ucp_worker_progress(sender_worker);
+            }
+            ASSERT_EQ(UCS_ERR_CANCELED, status);
 
-    /* Close sender EP to force send operation to complete with CANCEL status */
-    ucp_worker_h sender_worker = sender().worker();
-    void *close_req = sender().disconnect_nb(0, 0, UCP_EP_CLOSE_MODE_FORCE);
-    if (UCS_PTR_IS_PTR(close_req)) {
-        while (ucp_request_check_status(close_req) == UCS_INPROGRESS) {
-            ucp_worker_progress(sender_worker);
+            ASSERT_NE((void*)NULL, ucp_memh);
+            ucp_mem_unmap(sender().ucph(), ucp_memh);
         }
+
+        status = request_wait(sreq);
+        ASSERT_EQ(UCS_ERR_CANCELED, status);
+
+        /* Receive buffer should not be updated */
+        ASSERT_EQ(recv_buf, recv_copy);
+        /* Update send buffer by new data - emulation of free(buffer) */
+        memset(&send_buf[0], 'z', size);
+
+        /* Complete receive operation */
+        rreq = ucp_tag_msg_recv_nb(receiver().worker(), &recv_buf[0], size,
+                                ucp_dt_make_contig(1), message, tag_recv_cb);
+        ASSERT_TRUE(UCS_PTR_IS_PTR(rreq));
+        status = request_wait(rreq);
+
+        /* Receive request should fail or data should be valid */
+        EXPECT_TRUE((status != UCS_OK) || (recv_buf == send_buf));
     }
 
-    status = request_wait(sreq);
-    ASSERT_EQ(UCS_ERR_CANCELED, status);
+    ucs::ptr_vector<ucs::scoped_setenv> m_env;
+};
 
-    /* Receive buffer should not be updated */
-    ASSERT_EQ(recv_buf, recv_copy);
-    /* Update send buffer by new data - emulation of free(buffer) */
-    memset(&send_buf[0], 'z', size);
 
-    /* Complete receive operation */
-    rreq = ucp_tag_msg_recv_nb(receiver().worker(), &recv_buf[0], size,
-                               ucp_dt_make_contig(1), message, tag_recv_cb);
-    ASSERT_TRUE(UCS_PTR_IS_PTR(rreq));
-    status = request_wait(rreq);
+UCS_TEST_P(test_ucp_sockaddr_protocols_err_sender, tag_rndv_killed_sender,
+           "RNDV_THRESH=10k", "RNDV_SCHEME=get_zcopy")
+{
+    do_tag_rndv_killed_sender_test(false);
```

Comment:
what does this test check?

Reply:
it does a lot of `EXPECT_*`/`ASSERT_*`

Reply:
isn't it a normal flow, probably covered by other tests?